### PR TITLE
Add csp rules for api

### PR DESCRIPTION
**Pull request by mingan666**
This adds some new CSP rules needed to be able to view the API docs.

### CODE REVIEW

**Review by ildyria — APPROVED** [review]
LGTM.

File: config/secure-headers.php [hunk 1]
```diff
@@ -558,6 +561,10 @@
 		'style-src' => [
 			'self' => true,
 			'unsafe-inline' => true, // We need this one due to direct styles (not just style classes) applied by JavaScript
+			'allow' => [
+				'https://unpkg.com/@stoplight/elements/styles.min.css',
+				// 'url',
```
Comment (d7415) [inline config/secure-headers.php]:
```suggestion
```

**Comment by d7415** [comment]
Is this only relevant for dev builds? If so, do we really want to add this noise for everyone?

**Comment by ildyria** [comment]
> Is this only relevant for dev builds? If so, do we really want to add this noise for everyone?

It is not much noise though.

But a full proper approach would be checking the route and disabling the CSP on the fly (see e.g. for log-viewer).
On this specific case, we are a bit less constrained and can directly white list 2 urls instead of doing a full disable.

**Comment by ildyria** [comment]
@d7415 better ?

**Comment by ildyria** [comment]
Made the pull request cleaner:
- no more hacky functions
- use middleware to disable the CSP when necessary
- enable CSP by default, disable programatically.

**Review by qwerty287 — APPROVED** [review]
Untested

**Review by d7415 — APPROVED** [review]
Only skimmed, but it feels neater than the previous options.

**Pull request merged**